core/ir/x86/decode.c

```diff
@@ -1761,6 +1761,10 @@ decode_modrm(decode_info_t *di, byte optype, opnd_size_t opsize, opnd_t *reg_opn
         }
         force_full_disp = !needs_full_disp && di->has_disp && disp >= INT8_MIN &&
             disp <= INT8_MAX && di->mod == 2;
+        if (di->evex_encoded) {
```

Comment:
May as well merge this into the end of the same if condition a few lines up at 1756-1761

Reply:
Or does the force_full_disp computation need to use the un-compressed-scaled disp?

Reply:
We can merge them. force_full_disp is actually mutually exclusive (because it requires di->mod == 2 and this EVEX compressed disp requires di->mod == 1).